pwiz_tools/Shared/CommonUtil/SystemUtil/HttpClientWithProgress.cs

```diff
@@ -1020,9 +1125,93 @@ public static bool IsNetworkReallyAvailable()
             return false;
         }
 
+        /// <summary>
+        /// Disposes this HttpClientWithProgress instance.
+        /// Note: The underlying HttpClient is a singleton and is never disposed.
+        /// This method clears instance state but does not dispose the shared HttpClient.
+        /// </summary>
         public void Dispose()
         {
-            _httpClient?.Dispose();
+            // Clear per-request state
+            _authHeader = null;
+            _customHeaders.Clear();
+            // Note: _cookieContainer is managed by the caller (e.g., HttpPanoramaRequestHelper)
+            // and should not be disposed here. The shared HttpClient is never disposed.
+        }
+    }
+
+    /// <summary>
+    /// Provides a singleton HttpClient instance that is shared across all HttpClientWithProgress instances.
+    /// The HttpClient is created on first use and lives until process termination to avoid thread handle leaks.
+    /// </summary>
+    internal static class HttpClientSingleton
+    {
+        private static readonly Lazy<HttpClient> _instance = new Lazy<HttpClient>(CreateHttpClient, LazyThreadSafetyMode.ExecutionAndPublication);
+        private static HttpClientHandler _handler;
+        private static bool _disposed;
+
+        static HttpClientSingleton()
+        {
+            // Register for process exit to dispose HttpClient cleanly
+            // This prevents background threads from blocking process shutdown
+            AppDomain.CurrentDomain.ProcessExit += (sender, e) => DisposeInstance();
```

Comment:
I wonder if this line of code should be inside of "CreateHttpClient".
There's no work to be done until after the client has been created, so it might as well be the CreateHttpClient which adds the ProcessExit event handler.

Reply:
Good clean-up suggestion.